frontend/public/components/catalog-items.jsx

```diff
@@ -1,48 +1,204 @@
+      };
+      if (activeCategory.subCategories) { // show list of sub-categories
+        _.assign(state, {
+          currentCategories: activeCategory.subCategories,
+          showAllItemsForCategory: null,
+        });
+      } else { // no sub-categories, show all items for selected category
+        _.assign(state, {
+          currentCategories: parentCategory ? currentCategories : this.allCategories,
+          showAllItemsForCategory: category,
+        });
+      }
+      this.setState(state);
+    }
+  }
+
+  onActivateTab(tabs) {
+    const parentCategory = tabs.length === 2 ? tabs[0] : null;
+    const category = tabs.length === 2 ? tabs[1] : tabs[0];
+    this.syncTabsAndTiles(category, parentCategory);
+  }
+
+  onViewAllTilesForCategory(category, parentCategory) {
+    this.syncTabsAndTiles(category, parentCategory);
+  }
+
+  renderCategoryTiles(category) {
+    const { showAllItemsForCategory } = this.state;
+
+    if (!showAllItemsForCategory || category.id === showAllItemsForCategory) {
+      return (
+        <CatalogTileView.Category
+          key={category.id}
+          title={category.label}
+          totalItems={category.items && category.items.length}
+          viewAll={showAllItemsForCategory === category.id}
+          onViewAll={() => this.onViewAllTilesForCategory(category.id, category.parentCategory)}
+        >
+          {category.items &&
+          category.items.map((item) => {
+            const catalogTile = <CatalogTile
+              id={item.obj.metadata.uid}
+              key={item.obj.metadata.uid}
+              title={item.tileName}
+              iconImg={item.tileImgUrl}
+              iconClass={item.tileIconClass ? `icon ${normalizeIconClass(item.tileIconClass)}` : null}
+              vendor={item.tileProvider ? `Provided by ${item.tileProvider}` : null}
+              description={item.tileDescription}
+            />;
+            return item.href ? <Link key={item.obj.metadata.uid} className="co-catalog-item-tile"
+              to={item.href}>{catalogTile}</Link> : catalogTile;
+          })}
+        </CatalogTileView.Category>
+      );
+    }
+
+    return null;
+  }
+
+  getCategoryLabel(categoryId) {
+    return _.find(this.allCategories, (category) => category.id === categoryId).label;
+  }
 
-export const CatalogList = ({items}) =>
-  <CatalogTileView>
-    <CatalogTileView.Category key="all" title="All">
-      {_.map(items, (item) => <CatalogItem key={item.obj.metadata.uid} item={item} />)}
-    </CatalogTileView.Category>
-  </CatalogTileView>;
+  render() {
+    const { activeTabs, showAllItemsForCategory, currentCategories, numItems } = this.state;
 
-CatalogList.displayName = 'CatalogList';
+    let activeCategory = showAllItemsForCategory ? _.find(currentCategories, (category) => category.id === showAllItemsForCategory) : null;
 
-CatalogList.propTypes = {
-  items: PropTypes.array.isRequired
+    return (
+      <div className="row">
+        <div className="col-sm-2">
```

Comment:
I don't believe we want a bootstrap grid here. The eventual use of the filterSidePanel will set the width of the sidebar. The outer div should be `display: flex` the main content should have `flex: 1`.